example/idp/main.go

```diff
@@ -148,12 +166,33 @@ func main() {
 	hydraConfig := helpers.NewHydraConfig(*configPath)
 
 	// Setup the providers
-	var err error
-	provider, err = basic.NewBasicAuth(*htpasswdPath, "localhost")
+	userdb, err := memory.NewMemStore()
+	if err != nil {
+		panic(err)
+	}
+
+	err = userdb.LoadHtpasswd(*htpasswdPath)
 	if err != nil {
 		panic(err)
 	}
 
+	if *useForm {
```

Comment:
Let's not mix those two examples (BasicAuthProvider and FormProvider). Leave this file unchanged and"
- create another directory with the form provider `./example/idp-form` and put the modified `main.go` there 
- create docker file next to the original one `./idp-form-dockerfile`
- add configuration in docker-compose like :

```
  idp-form:
    container_name: idp-form
    dockerfile: idp-form-dockerfile
    build:
      context: ../
    environment:
      - HYDRA_URL=https://hydra:4444
    links:
      - hydra
    volumes:
      - hydravolume:/root
    ports:
      - "3000:3000"

```